README.md
# Lab Members

Lab Members is a Django app to display lab personnel and information about them.

<!-- Detailed documentation is in the "docs" directory. -->

## Quick start

- Edit the project's `settings.py` file.

    - Add `lab_members` and its dependencies to your `INSTALLED_APPS` setting:

        ```python
        INSTALLED_APPS = (
            ...
            'lab_members',
            'easy_thumbnails',
            'filer',
            'mptt',
            'sekizai',
            'friendlytagloader',
        )
        ```

    - Specify your media settings, if not already specified:

        ```python
        MEDIA_URL = '/media/'
        MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
        ```

    - Add `filer` and `easy_thumbnail` settings: 

        ```python
        # For filer's Django 1.7 compatibility
        MIGRATION_MODULES = {
            ...
            'filer': 'filer.migrations_django',
        }

        # For easy_thumbnails to support retina displays (recent MacBooks, iOS)
        THUMBNAIL_HIGH_RESOLUTION = True
        THUMBNAIL_QUALITY = 95
        THUMBNAIL_PROCESSORS = (
            'easy_thumbnails.processors.colorspace',
            'easy_thumbnails.processors.autocrop',
            'filer.thumbnail_processors.scale_and_crop_with_subject_location',
            'easy_thumbnails.processors.filters',
        )
        THUMBNAIL_PRESERVE_EXTENSIONS = ('png', 'gif')
        THUMBNAIL_SUBDIR = 'versions'
        ```

    - Add `sekizai` settings:

        ```python
        from django.conf.global_settings import TEMPLATE_CONTEXT_PROCESSORS
        TEMPLATE_CONTEXT_PROCESSORS += ('sekizai.context_processors.sekizai',)
        )
        ```

- Include URL configurations for `lab_members` and media (if `DEBUG == True`) in your project's `urls.py` file:

    ```python
    ...
    from django.conf import settings

    urlpatterns = patterns('',
        ...
        url(r'^lab_members/', include('lab_members.urls', namespace='lab_members')),
        ...
    )

    if settings.DEBUG:
        urlpatterns += patterns('',
            (r'^media/(?P<path>.*)$', 'django.views.static.serve', {
            'document_root': settings.MEDIA_ROOT}))
    ```

- Run `python manage.py makemigrations lab_members` to create the lab_members migrations.

- Run `python manage.py migrate` to create the lab_members models.

- Start the development server (`python manage.py runserver`) and visit http://127.0.0.1:8000/admin/
   to create a lab member (you'll need the Admin app enabled).

- Visit http://127.0.0.1:8000/lab_members/ to view a list of lab members.

*Version 0.2.5*


CHANGELOG
Revision history for Django app django-lab-members

0.2.5 2015-05-22

  - Add Scientist publications placeholder
  - Add Admin models for Education/Employment Records
  - Fix ordering of Education/Employment Records

0.2.4 2015-05-08

  - Display Scientist only if visible == True
  - Prevent higher resolution photos from being wider than 300px
  - Fix linking to Lab Alumni subsection
  - Validate that Employment/Education 'Year Started' <= 'Year Ended'
  - Open advisor's link in a new window
  - Make alumni link more prominent
  - Add website field for Scientist & display below email address
  - Move sidebar to modal for smaller screens
  - Minor behind-the-scenes improvements

0.2.3 2015-04-29

  - Make profile thumbnails equal heights (within row) to account for long names
  - Use 'django-friendly-tag-loader' to load and use django CMS tags if they are available
  - Use 2x resolution photos (for retina, etc. displays)

0.2.2 2015-04-21

  - Add Scientist sidebar placeholder if CMS for medium+ screens
  - Improve styling

0.2.1 2015-04-19

  - Improve styling

0.2.0 2015-04-10

  - Allow integration with django CMS using djangocms-lab-members

0.1.0 2015-04-08

  - A Django app to display lab personnel and information about them

django_lab_members_site/settings.py
"""
Django settings for django_lab_members_site project.

For more information on this file, see
https://docs.djangoproject.com/en/1.7/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.7/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os
BASE_DIR = os.path.dirname(os.path.dirname(__file__))


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/1.7/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'pr4easmqn356(1c9h&9vs!pdgztgy=@*wayorsp*4_w)4!@r+#'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

TEMPLATE_DEBUG = True

ALLOWED_HOSTS = []


# Application definition

DEFAULT_APPS = (
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
)

THIRD_PARTY_APPS = (
    'easy_thumbnails',
    'filer',
    'mptt',
)

LOCAL_APPS = (
    'lab_members',
)

INSTALLED_APPS = DEFAULT_APPS + THIRD_PARTY_APPS + LOCAL_APPS

MIDDLEWARE_CLASSES = (
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
)

ROOT_URLCONF = 'django_lab_members_site.urls'

WSGI_APPLICATION = 'django_lab_members_site.wsgi.application'


# Database
# https://docs.djangoproject.com/en/1.7/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
    }
}

# Internationalization
# https://docs.djangoproject.com/en/1.7/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'America/Los_Angeles'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.7/howto/static-files/

STATIC_URL = '/static/'
MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# For filer's Django 1.7 compatibility
MIGRATION_MODULES = {
    'filer': 'filer.migrations_django',
}

# For easy_thumbnails to support retina displays (recent MacBooks, iOS)
THUMBNAIL_HIGH_RESOLUTION = True
THUMBNAIL_QUALITY = 95
THUMBNAIL_PROCESSORS = (
    'easy_thumbnails.processors.colorspace',
    'easy_thumbnails.processors.autocrop',
    #'easy_thumbnails.processors.scale_and_crop',
    'filer.thumbnail_processors.scale_and_crop_with_subject_location',
    'easy_thumbnails.processors.filters',
)
THUMBNAIL_PRESERVE_EXTENSIONS = ('png', 'gif')
THUMBNAIL_SUBDIR = 'versions'
